scripts/list-routes.rb

```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env ruby
+
+require 'json'
+require 'set'
+
+routes_by_domain_url = {}
+
+page = JSON.load `cf curl '/v2/routes?results-per-page=100'`
+next_url = page["next_url"]
+
+while next_url
+  page = JSON.load `cf curl '#{next_url}'`
```

Comment:
This reads like it accidentally discards the first page of results. Setting `next_url = '/v2/routes?results-per-page=100'` would workaround that.

Reply:
Good catch - I'm just making deliberate mistakes to keep you on your toes. 🤦‍♂

Reply:
It's always nice to know that code review can catch bugs :)

Reply:
I'll re-review after lunch 🥖